include/cuco/detail/open_addressing_ref_impl.cuh

```diff
@@ -850,12 +861,12 @@ class open_addressing_ref_impl {
    *
    * @return Result of this operation, i.e., success/continue/duplicate
    */
-  template <bool HasPayload, typename Predicate>
+  template <bool HasPayload, typename Value, typename Predicate>
   [[nodiscard]] __device__ insert_result attempt_insert(value_type* slot,
-                                                        value_type const& value,
+                                                        Value const& value,
                                                         Predicate const& predicate) noexcept
   {
-    if constexpr (sizeof(value_type) <= 8) {
+    if constexpr ((sizeof(value_type) <= 8) and cuda::std::is_convertible_v<Value, value_type>) {
```

Comment:
This is kinda weird.
In a `map` scenario, a user could pass a custom pair such as `pair<wrapper<Key>, wrapper<Payload>>`, where `wrapper<T>` is implicitly convertible to `T`. However, that doesn't imply that the pair itself is implicitly convertible to `pair<Key, Payload>`, i.e., the `value_type` of the map. Hence, we can't use `packed_cas` in this scenario (I think). The only way to enable `packed_cas` would be to explicitly re-pack the pair, i.e., manually convert `pair<wrapper<Key>, wrapper<Payload>> -> pair<Key, Payload>` before passing it to `packed_cas`.

Reply:
> However, that doesn't imply that the pair itself is implicitly convertible to pair<Key, Payload>

Is `sizeof(value_type) <= 8` still valid in this case?

Reply:
This is the moment we need to construct the slot content to insert from the user-provided insert pairs.